pydicom/valuerep.py

```diff
@@ -894,6 +894,329 @@ def __bool__(self) -> bool:
+            # for different parts of the input
+            if isinstance(val, bytes):
+                val_enc = val
+                val_dec = dec(val)
+            else:
+                val_enc = enc(val)
+                val_dec = val
+
+            # Check for disallowed chars in the decoded string
+            for c in disallowed_chars:
+                if c in val_dec:
+                    raise ValueError(
+                        f'Strings may not contain the {c} character'
+                    )
+
+            # Return the encoded string
+            return val_enc
+
+        def make_component_group(components: List[Union[str, bytes]]):
+            encoded_components = map(standardize_encoding, components)
+            joined_components = encoded_component_sep.join(encoded_components)
+            return joined_components.rstrip(encoded_component_sep)
+
+        component_groups = [
+            make_component_group(alphabetic_group),
+            make_component_group(ideographic_group),
+            make_component_group(phonetic_group)
+        ]
+        joined_groups = encoded_group_sep.join(component_groups)
+        joined_groups = joined_groups.rstrip(encoded_group_sep)
+        return joined_groups
+
+    @classmethod
+    def from_named_components(
+        cls,
+        family_name: Union[str, bytes] = '',
+        given_name: Union[str, bytes] = '',
+        middle_name: Union[str, bytes] = '',
+        name_prefix: Union[str, bytes] = '',
+        name_suffix: Union[str, bytes] = '',
+        family_name_ideographic: Union[str, bytes] = '',
+        given_name_ideographic: Union[str, bytes] = '',
+        middle_name_ideographic: Union[str, bytes] = '',
+        name_prefix_ideographic: Union[str, bytes] = '',
+        name_suffix_ideographic: Union[str, bytes] = '',
+        family_name_phonetic: Union[str, bytes] = '',
+        given_name_phonetic: Union[str, bytes] = '',
+        middle_name_phonetic: Union[str, bytes] = '',
+        name_prefix_phonetic: Union[str, bytes] = '',
+        name_suffix_phonetic: Union[str, bytes] = '',
+        encodings: Optional[List[str]] = None,
+    ) -> 'PersonName':
+        """Construct a PersonName from explicit named components.
+
+        The DICOM standard describes human names using five components:
+        family name, given name, middle name, name prefix, and name suffix.
+        Any component may be an empty string (the default) if not used.
+        A component may contain multiple space-separated words if there
+        are, for example, multiple given names, middle names, or titles.
+
+        Additionally, each component may be represented in ideographic or
+        phonetic form in addition to (or instead of) alphabetic form.
+
+        For more information see the following parts of the DICOM standard:
+        - `Value Representations <http://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_6.2.html>`_
```

Comment:
You can shorten these with (for example)
```python
:dcm:`Section 6.2<part05/sect_6.2.html>`
```

Reply:
Neat! Out of interest how did you configure sphinx to do this?

Addressed with 7324a6a

Reply:
The sphinx.ext.extlinks package lets you do it, you just define [your own shortcuts](https://github.com/pydicom/pydicom/blob/master/doc/conf.py#L86) in conf.py